core/src/com/unciv/logic/city/CityConstructions.kt

```diff
@@ -581,6 +581,23 @@ class CityConstructions : IsPartOfGameInfoSerialization {
         updateUniques()
     }
 
+    fun removeBuildings(buildings: Set<Building>) {
+        val newBuildings = ArrayList<Building>(builtBuildingObjects.size - buildings.size)
```

Comment:
This assumes every building passed into this function is in the builtBuildingsObjects, which means if the city actually doesn't have the building, the empty ArrayList might be smaller than expected. Giving this a starting size seems unnecessary here (and is largely ignored), having it in the first place seems unnecessary and misleading

Reply:
I guess. Even still, it assumes that it'll always be called in a situation where we know the buildings are actually there. None of the other lines in this function really care about that, though, so as baseline I almost feels like it should intuitively either just not bother with an initial capacity (consistent with how we handle it anyways. Premature optimization...) or actually check how many buildings that will be left over when we're done and set the initial capacity correctly